test/stdlib/Random.swift

```diff
@@ -0,0 +1,246 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+let RandomTests = TestSuite("Random")
+
+// _stdlib_random
+
+RandomTests.test("_stdlib_random") {
+  for count in [100, 1000] {
+    var bytes1 = [UInt8](repeating: 0, count: count)
+    var bytes2 = [UInt8](repeating: 0, count: count)
+    let zeros = [UInt8](repeating: 0, count: count)
+    expectEqual(bytes1, bytes2)
+    expectEqual(bytes1, zeros)
+    expectEqual(bytes2, zeros)
+    
+    bytes1.withUnsafeMutableBytes { _stdlib_random($0) }
+    expectNotEqual(bytes1, bytes2)
+    expectNotEqual(bytes1, zeros)
+    
+    bytes2.withUnsafeMutableBytes { _stdlib_random($0) }
+    expectNotEqual(bytes1, bytes2)
+    expectNotEqual(bytes2, zeros)
+  }
+}
+
+// Basic random numbers
+
+RandomTests.test("basic random numbers") {
```

Comment:
Could this also test the full ranges (min to max) of `Int8` and `UInt8`?

The number of iterations would need to increase (to 100_000 or 1_000_000 perhaps).

Reply:
I'm not sure I'm following when you say "test the full ranges".

Reply:
Sorry, I attached the previous comment to the wrong test. I meant the "random integers in ranges" test, using the `integerRangeTest` helper function. It's currently testing for the min/max ± 10, but it might be good to also test for the entire 256 elements of the 8-bit closed ranges (and maybe half-open ranges).

* `(Int8.min ... .max)` and `(UInt8.min ... .max)`
* `(Int8.min ..< .max)` and `(UInt8.min ..< .max)`

Reply:
I've done a few tests with 10_000 iterations and it hasn't failed once, so I'm going to set it to that initially. Let me know if we still need to increase it just in case.

Reply:
You might be able to simplify `integerRangeTest` by adding a couple more helper functions (one taking a half-open range, another taking a closed range). The number of iterations could then be `for _ in 0 ..< (100 * range.count)` for example. The body of the existing `integerRangeTest` would be:

```swift
integerRangeTest(in: T.min ..< (T.min + 10))
integerRangeTest(in: T.min ... (T.min + 10))
integerRangeTest(in: (T.max - 10) ..< T.max)
integerRangeTest(in: (T.max - 10) ... T.max)
if T.bitWidth == 8 {
  integerRangeTest(in: T.min ..< T.max)
  integerRangeTest(in: T.min ... T.max)
}
```

By the way, all the recent changes have been squashed into the "Consolidate `_stdlib_random` functions" commit.